app/src/main/java/com/example/android/receitaboa/activity/NovaReceitaFotoActivity.java

```diff
@@ -285,21 +243,70 @@ private void mensagemFotoAdicionada() {
     }
 
 
-    private File salvarImagemDiretorio() throws IOException {
+    private void salvarImagemDirCelular(Bitmap imgFoto) throws IOException {
+
+        File filePath = Environment.getExternalStorageDirectory();
```

Comment:
Existe alguma razão para você guardar o arquivo no ExternalStorageDirectory? Esse método é para guardar arquivos num cartão SD ou na memória externo, fora do domínio app. Isso só é útil para arquivos que o usuário queira usar depois. Estes dados não são deletados quando você delete o app ou quando você limpa a cache e exigem permissão de escrita. Pior, pode haver celulares que não tem acesso ao armazenamento externo e no caso deles vai dar erro na hora de tentar criar o arquivo

Você deveria usar getFilesDir(), ele te retorna um diretório dentro do sandbox do próprio app, não exige permissão de escrita e quando você limpa o cache do app os arquivos são apagados juntos. Se não me engano, usando este método, pode até apagar a permissão de escrita que você adicionou. Tente sempre usar o mínimo de permissões possíveis para o teu app

Reply:
pensei que era a memória interna. Vou trocar.

Reply:
não sei se eu li direito, mas na internet estão falando que o getExternalStorageDirectory é a memória interna para imagens.

Reply:
Environment.getExternalStorageDirectory().getAbsolutePath()
Gives you the full path the SDCard. You can then do normal File I/O operations using standard Java.

Imagino que as pessoas tão sugerindo para guardar fotos no cartão SD. No caso, você também tá salvando fotos, mas são fotos que são de interesse interno do app, então é arquivo interno do app (da mesma forma que um ícone ou um thumbnail do Youtube, não é algo que você vai querer largar no cartão SD do usuário para encher memória que depois o usuário não consegue deletar com facilidade)